db/db_impl.cc

```diff
@@ -182,7 +182,7 @@ DBImpl::DBImpl(const DBOptions& options, const std::string& dbname,
       has_unpersisted_data_(false),
       unable_to_flush_oldest_log_(false),
       env_options_(BuildDBOptions(immutable_db_options_, mutable_db_options_)),
-      env_options_for_compaction_(env_->OptimizeForCompactionTableWrite(
+      env_options_for_compaction_read_(env_->OptimizeForCompactionTableWrite(
```

Comment:
Sorry, looks like this variable is used by both compaction read and write. Let's revert its name back to `env_options_for_compaction_`. My bad.